### ISSUE TITLE
Reprojection artifacts, bbox-crs extents

### ISSUE THREAD

**Issue by IvanSanchez** [opengeospatial/ogcapi-maps#85]
I'm writing this in the 2022-03-08/10 OGC/OSGeo code sprint, at suggested by @jerstlouis . Might be related to https://github.com/opengeospatial/ogcapi-maps/issues/82.

I've been working on two (very very similar) client implementations - one in https://gitlab.com/IvanSanchez/leaflet.imageoverlay.ogcapi , the second one in https://gitlab.com/IvanSanchez/gleo/-/commit/82f041d0dd189dfb9f67b7adc38a98044917dffa .

The issue is that, when requesting a map natively stored in EPSG:4326 but requested in EPSG:3857, the behaviour is buggy/unclear when the EPSG:3857 bounding box goes beyond the *de facto* extents of EPSG:3857 stuff (the +/-85° meridian).

So when scrolling a map north, beyond +85° (or 20037508.34 northing units in EPSG:3857), the result looks like:

![image](https://user-images.githubusercontent.com/1125786/157758987-55fafaac-566a-4b3c-8a7e-9a4de1511b17.png)

The URL for the image is `https://maps.ecere.com/ogcapi/collections/blueMarble/map.png?crs=EPSG%3A3857&bbox-crs=EPSG%3A3857&bbox=-15293005.97236237%2C1118504.0254117623%2C11084495.240840755%2C29296250.12853676&width=337&height=360&transparent=false`, or 

```
https://maps.ecere.com/ogcapi/collections/blueMarble/map.png
?crs=EPSG:3857
&bbox-crs=EPSG:3857
&bbox=-15293005.97236237,1118504.0254117623,11084495.240840755,29296250.12853676
&width=337
&height=360
&transparent=false
```

Note the maximum northing of 29296250.12 - higher than the implicit 20037508.34 extent.

The image by itself looks like:

![image](https://user-images.githubusercontent.com/1125786/157759312-6def748d-5711-4ed4-ab3f-9c54eacac731.png)

...note how the image is stretched vertically, even when the aspect ratio between the requested bbox and the image width/height is the correct one.

I suspect this is purely wrong behaviour on the reprojection being performed on the server implementation. I'm not sure if this has any impact on the actual API specification.

By comparison, the client implementation at https://ivansanchez.gitlab.io/leaflet.imageoverlay.ogcapi/demo.html looks right **but the client implementation itself is wrong**: Leaflet is reporting a north latitude higher than +85°, but can only place geometries within the `-20037508.34,+20037508.34` extents. Therefore, the client is *wrongly* squashing an image that the server is *wrongly* stretching.

**Comment by jerstlouis** [comment]
@IvanSanchez What happens here is that our server refuses to honor the requested BBOX because it is beyond the maximum latitude values defined for EPSG:3857 (±85.08405903), therefore it clips the BBOX to the requested extent. Technically it is impossible to return the requested extent unless some assumptions are made, as EPSG:3857 is undefined outside of those bounds. I think the only other option for the server would have been to return a 400 _client error_ HTTP status / invalid parameter value explanation.

The width and height parameter were understood as-is without any pro-rated adjustment. If only one of the two is specified, the other would have been automatically computed respecting the aspect ratio.

It should probably be clarified whether and when a server is allowed to return a different extent than requested.
In the case of requesting a BBOX larger than the available data (but within the output CRS defined bounds), I think it would be preferable to at least recommend that servers honor the requested BBOX even if the extent is larger than the available data (and filled with background color and/or transparency), because that is a use case for cutting out map tiles, for example.

In this particular case, I am of the opinion that what our server is currently doing is reasonable.
However, the issue is related to returning an image for which wrong georeferencing is assumed.
No such stretching occurs e.g. if a [GeoTIFF](https://maps.ecere.com/ogcapi/collections/blueMarble/map.tiff?crs=EPSG%3A3857&bbox-crs=EPSG%3A3857&bbox=-15293005.97236237%2C1118504.0254117623%2C11084495.240840755%2C29296250.12853676&width=337&height=360&transparent=false) is requested instead, and the response is visualized in QGIS.
This issue of unknown bounding box is also an issue with the default parameter (not specifying any bbox), or when specifying a `bbox-crs` in a different CRS than the output crs (`crs`).

As we did for _OGC API - Tiles_ [datetime](http://docs.ogc.org/DRAFTS/20-057.html#_actual_date_time_response_header) conformance class to return the actual date & time of the response (`OGCAPI-datetime`), and in [_OGC API - Features - Part 2: CRS by reference_ ](https://docs.ogc.org/is/18-058/18-058.html#_coordinate_reference_system_information_independent_of_the_feature_encoding) (`Content-Crs:`), whose functionality we hope to make [_Common_](https://github.com/opengeospatial/ogcapi-tiles/tree/master/openapi/parameters/common-crs) across different OGC API specifications including _Maps_, we had suggested to include an HTTP response header to unambiguously indicate indicate the bounding box of the response, e.g. `OGCAPI-bbox`, in the CRS of the **response** (`crs` query parameter, or default).
